lib/manageiq_performance/middlewares/active_support_timers.rb

```diff
@@ -19,6 +19,14 @@ def active_support_timers_initialize
       def active_support_timers_start(env); end
 
       def active_support_timers_finish(env)
+        if Thread.current[:miq_perf_request_timer_data].nil? and env["HTTP_MIQ_PERF_TIMESTAMP"]
+          time = (env["MIQ_PERFORMANCE_END_TIME"] - env["HTTP_MIQ_PERF_TIMESTAMP"]) / 1000
```

Comment:
Since the `env["HTTP_MIQ_PERF_TIMESTAMP"]` comes from a `HTTP` header, we should run a `.to_i` on it just to make sure it is an `int` and we aren't trying to subtract a string.